src/types/box-types.ts

```diff
@@ -34,11 +45,8 @@ export type BoxProps<T extends Is> = &
   }
 
 export interface BoxComponent<T extends Is> {
-  // This is the desired type (inspired by reakit)
-  // <TT extends Is = T>(props: BoxProps<TT>): JSX.Element
-  // Unfortunately, TypeScript doesn't like it. It works for string elements
-  // and functional components without generics, but it breaks on generics.
-  // The following two types are a workaround.
-  <TT extends Is>(props: BoxProps<TT> & { is?: TT }): React.ReactElement | null
-  (props: BoxProps<T>): React.ReactElement | null
+  <TT extends Is = T>(props: BoxProps<TT>): React.ReactElement | null
+  propTypes?: any
+  defaultProps?: any
```

Comment:
Won't these break the React types (by removing any type checking)?

Reply:
This is actually saving us the trouble of casting them `as any` in `box.tsx`.

It's no better or worse as for types, except that we don't have to cast. I'm not sure what React types would break?

This isn't using `React.FunctionComponent` or `React.ClassComponent`.

Reply:
I mean for any component inheriting the Box props like in Evergreen? Will it break type checking of `propTypes` and `defaultProps` for components using `interface Props extends React.ComponentProps<typeof Box> {}`?

Reply:
Ah, good question. I don't think so, but I"ll test it out.

Reply:
It works for everything except the `is` prop which seems to be due to something else (not `defaultProps`)

![Screen Shot 2019-06-06 at 1 29 21 PM](https://user-images.githubusercontent.com/710752/59057127-53295880-885f-11e9-8eb8-27a8df94cb46.png)
![Screen Shot 2019-06-06 at 1 29 05 PM](https://user-images.githubusercontent.com/710752/59057128-53295880-885f-11e9-92a7-588aa1fbb536.png)
![Screen Shot 2019-06-06 at 1 31 16 PM](https://user-images.githubusercontent.com/710752/59057158-64726500-885f-11e9-9a12-ad3c25aa8dd7.png)

Reply:
It doesn't seem to affect checking `propTypes` or `defaultProps` but I do see that this typedef doesn't work when you `extend React.ComponentProps<typeof Box>` specifically the `is` prop doesn't get extended properly.

Reply:
![Screen Shot 2019-06-17 at 11 43 57 AM](https://user-images.githubusercontent.com/710752/59621576-36a9dd80-90f5-11e9-93e9-9fb72d065c10.png)

I think `extends` doesn't work with the `is` prop because we are not using the stock `FunctionComponent` definition for `BoxComponent` in order to get the inference working right...